test/unit-tests/function/numeric/solveODE.test.js

```diff
@@ -0,0 +1,189 @@
+import assert from 'assert'
+import math from '../../../../src/defaultInstance.js'
+
+const solveODE = math.solveODE
+const matrix = math.matrix
+const subtract = math.subtract
+const exp = math.exp
+const min = math.min
+const max = math.max
+const abs = math.abs
+const tol = 1e-3
+const smaller = math.smaller
+const smallerEq = math.smallerEq
+const largerEq = math.largerEq
+const multiply = math.multiply
+const dotMultiply = math.dotMultiply
+const divide = math.divide
+const diff = math.diff
+const unit = math.unit
+const map = math.map
+const transpose = math.transpose
+
+function withinTolerance (A, B, tol) {
+  const maxAbsError = max(abs(subtract(A, B)))
+  return smaller(maxAbsError, tol)
+}
+
+describe('solveODE', function () {
```

Comment:
There are quite some tests already, great! Can you add some unit tests testing:

1. `BigNumber` input values
2.  Passing a `Matrix` instead of an `Array` (that is not yet supported but will be very useful since the expression parser by default creates a `Matrix`).

Reply:
The test for Matrix is implemented, now it return matrices when the inputs are matrices

Regarding big numbers, I've been trying to make them work but I keep getting 

`TypeError: Cannot implicitly convert a number with >15 significant digits to BigNumber (value: 0.3333333333333333). Use function bignumber(x) to convert to BigNumber.`

I will try to fix it but I might need further review please. I included the test in comments

Reply:
Ah, that is most likely because we're mixing numbers like `-25360 / 2187 = -11.595793324188385` with big numbers. To solve that, those irrational numbers should be converted to BigNumber explicitly, like: 

```
a = [...]
aBigNumber = bignumber(a)
``` 

Maybe it is possible to create util functions `getButcherTableau(convert): { a, c, b, bp }`, where `convert` can be either `n => n` or `n => bignumber(n)`, so we can let it return either number or BigNumber values?

Reply:
Yes, thank you, this can be resolved.

You got it exactly.

One issue was with the `butcherTableau` and another with `delta`. I ended up implemented that logic with ternary operators but it seems also nice to use `convert`

I should have responded in here when I fixed the issue, I'm sorry about that.